### PR TITLE
Make ConvertRaster::getOffset() compatible with new Java versions

### LINKED ISSUE #354: An illegal reflective access operation has occurred
When using boocv for similarity search, it throws the following warning:
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by boofcv.io.image.ConvertRaster (file:/C:/Users/brandl/.gradle/caches/modules-2/files-2.1/org.boofcv/boofcv-io/0.40.1/a99b6dbd3e07646ebaecc715aed48a0f03596db0/boofcv-io-0.40.1.jar) to method sun.awt.image.ByteInterleavedRaster.getDataOffset(int)
WARNING: Please consider reporting this to the maintainers of boofcv.io.image.ConvertRaster
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

### PR DESCRIPTION
**Pull request by GenieTim**
Dear Mr. Abeles,

Thank you again for this incredibly useful library. 
With this PR, I hope to fix #354 as discussed therein. 
According to the tests and my local useage of the library, this change works and is sufficient. 
I could not determine a significant performance difference.

Best regards,

### CODE REVIEW

**Comment by lessthanoptimal** [comment]
Needs a few unit tests updated. 

- Remove line 102 in TestConvertRaster
- ConvertBufferedImage.isSubImageLegal() should return true now

The test in ConvertRaster works but other tests in TestConvertBufferedImage fail.

**Comment by lessthanoptimal** [comment]
image

The fix doesn't seem to work with 3-band byte array BufferedImages since each band has a unique offset. See the screen shot above from the debugger.

**Comment by GenieTim** [comment]
Thanks for the feedback! I see, it is not as simple as I hoped it would be. I will be working on it!

**Comment by GenieTim** [comment]
The next iteration I would propose is pushed; it uses the actual computations the JDK does to set the data offset. 

Now, since there are multiple JDK versions out there, I am not sure how far this implementation is compatible. I would guess not too bad, but use it as a fallback, just in case.

File: main/boofcv-io/src/main/java/boofcv/io/image/ConvertRaster.java [hunk 1]
```diff
@@ -273,13 +274,27 @@ public static int getOffset( WritableRaster raster ) {
...
 				min = Math.min(min, (Integer)m.invoke(raster, i));
 			}
 			return min;
-		} catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+		} catch (NoSuchMethodException | IllegalAccessException | InaccessibleObjectException | InvocationTargetException e) {
+
+			SampleModel sm = raster.getSampleModel();
+			if (sm instanceof ComponentSampleModel) {
+				ComponentSampleModel csm = (ComponentSampleModel)sm;
+				// according to https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/sun/awt/image/ByteInterleavedRaster.java
```
Comment (GenieTim) [inline main/boofcv-io/src/main/java/boofcv/io/image/ConvertRaster.java:289]: Interestingly, as of writing this comment, the linked implementation has `getDataOffset` declared as a public method.

**Comment by lessthanoptimal** [comment]
Can you squash this PR? Going to take a closer look at it but since the tests pass it should be good.

**Comment by GenieTim** [comment]
I _think_ you can squash and merge it as seen in the screenshot below, whereas I would have to either overwrite my history with just one commit, or submit a new PR with the changes in a new branch or something.

select-squash-and-merge-from-drop-down-menu

**Comment by lessthanoptimal** [comment]
ended up doing the merge into SNAPSHOT manually since I needed to add another minor change. Thanks for the PR!

**Comment by GenieTim** [comment]
Thank you for merging! Any tip on when to expect a new release?

**Comment by lessthanoptimal** [comment]
within in 1 or 2 weeks. Decided to sneak in a couple of new stereo related features at the last minute. Making sure that's stable first.

**Pull request closed without merging**